migrator/migrations/m_200_to_m_201_compliance_v2_for_4_5/migration_impl.go

```diff
@@ -0,0 +1,303 @@
+	query := db.WithContext(database.DBCtx).Table(newSchema.ComplianceOperatorProfileV2TableName).Select("serialized")
+
+	rows, err := query.Rows()
+	if err != nil {
+		return errors.Wrapf(err, "failed to iterate table %s", newSchema.ComplianceOperatorProfileV2TableName)
+	}
+	defer func() { _ = rows.Close() }()
+
+	var convertedProfiles []*newSchema.ComplianceOperatorProfileV2
+	var count int
+	for rows.Next() {
+		var profile *newSchema.ComplianceOperatorProfileV2
+		if err = query.ScanRows(rows, &profile); err != nil {
+			return errors.Wrap(err, "failed to scan rows")
+		}
+
+		profileProto, err := newSchema.ConvertComplianceOperatorProfileV2ToProto(profile)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert %+v to proto", profile)
+		}
+
+		// Add the profile ref id
+		profileProto.ProfileRefId = createProfileRefID(profileProto.GetClusterId(), profileProto.GetProfileId(), profileProto.GetProductType())
+
+		converted, err := newSchema.ConvertComplianceOperatorProfileV2FromProto(profileProto)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert from proto %+v", profileProto)
+		}
+		convertedProfiles = append(convertedProfiles, converted)
+		count++
+
+		if len(convertedProfiles) == batchSize {
+			// Upsert converted profiles
+			if err = db.Clauses(clause.OnConflict{UpdateAll: true}).Model(newSchema.CreateTableComplianceOperatorProfileV2Stmt.GormModel).Create(&convertedProfiles).Error; err != nil {
+				return errors.Wrapf(err, "failed to upsert converted %d profiles after %d upserted", len(convertedProfiles), count-len(convertedProfiles))
+			}
+			convertedProfiles = convertedProfiles[:0]
+		}
+	}
+
+	if err := rows.Err(); err != nil {
+		return errors.Wrapf(err, "failed to get rows for %s", newSchema.ComplianceOperatorProfileV2TableName)
+	}
+
+	if len(convertedProfiles) > 0 {
+		if err = db.Clauses(clause.OnConflict{UpdateAll: true}).Model(newSchema.CreateTableComplianceOperatorProfileV2Stmt.GormModel).Create(&convertedProfiles).Error; err != nil {
+			return errors.Wrapf(err, "failed to upsert last %d profiles", len(convertedProfiles))
+		}
+	}
+	log.Infof("Converted %d profile records", count)
+
+	return nil
+}
+
+func migrateRules(database *types.Databases) error {
+	db := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, db, newSchema.CreateTableComplianceOperatorRuleV2Stmt)
+	db.WithContext(database.DBCtx).Table(newSchema.ComplianceOperatorRuleV2TableName)
+
+	// Need to use store because of `RuleControls`
+	// Since we are only using the walk to retrieve data, we can used the updated store to retrieve the data and update it.
+	ruleStore := rulesStore.New(database.PostgresDB)
+
+	ruleCount := 0
+	rules := make([]*storage.ComplianceOperatorRuleV2, 0)
+	err := ruleStore.Walk(database.DBCtx, func(obj *storage.ComplianceOperatorRuleV2) error {
+		obj.ParentRule = obj.GetAnnotations()[v1alpha1.RuleIDAnnotationKey]
```

Comment:
Do we need to check if annotation exists? Same for other places where there is a key lookup on annotations.

Reply:
1.  That should never happen.
2. sure I can put some safety checks in there, just in case.  I would probably just skip those records as annotations are pretty important.

Reply:
actually I take that back.  If the annotation does not exist it will simply assign the 0 value.  In this case the empty string.  I will take a look through, but my inclination is no action needed.

Reply:
Sounds good